### PR TITLE
Fix Issue 16380 - no bindings for err.h

### PR DESCRIPTION
**Pull request by ErnyTech**
err.h although not present in the posix standard it is however present
in most distributions based on Darwin, BSD and Linux

### CODE REVIEW

**Comment by thewilsonator** [comment]
Seems fine to me, I'll wait for others to chime in though.

File: src/core/sys/darwin/err.d [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+void verrc(int eval, int code, scope const char* fmt, va_list args);
+void verrx(int eval, scope const char* fmt, va_list args);
+void vwarn(scope const char* fmt, va_list args);
+void vwarnc(int code, scope const char* fmt, va_list args);
+void vwarnx(scope const char* fmt, va_list args);
+void err_set_file(void* vfp);
+void err_set_exit(ExitFunction exitf);
+
+// ^ __OSX_AVAILABLE_STARTING(__MAC_10_6, __IPHONE_3_2);
+err_set_exit_b(ExitFunction exitf);
```
Comment (jacob-carlborg) [inline src/core/sys/darwin/err.d]: `err_set_exit_b` takes a block not a function pointer. A block is an Apple/LLVM extension to C, which is not representable in D.

**Comment by ErnyTech** [comment]
@jacob-carlborg so i need remove err_set_exit_b?

**Comment by jacob-carlborg** [comment]
Yes.

**Comment by jacob-carlborg** [comment]
It’s technically possible to create a struct in D with the same layout as a block, but that was not accepted in the past.

**Comment by ErnyTech** [comment]
Also on opensolaris there is the same thing
https://repo.or.cz/opensolaris.git/blob_plain/HEAD:/usr/src/head/err.h

**Pull request merged**